src/dsa/rpo_falcon512/signature.rs

```diff
@@ -26,10 +28,18 @@ use crate::utils::string::ToString;
 /// 2. 896 bytes for the public key. This is decoded into the `h` polynomial above.
 ///
 /// The actual signature is serialized as:
-/// 1. 40 bytes for the nonce.
-/// 2. 625 bytes encoding the `s2` polynomial above.
+/// 1. 1 header byte specifying the algorithm used to encode the coefficients of the `s2` polynomial
+///    together with the degree of the irreducible polynomial phi.
+///    The general format of this byte is 0b0cc1nnnn where:
+///     a. cc is either 01 when the compressed encoding algorithm is used and 10 when the
+///     uncompressed algorithm is used.
+///     b. nnnn is log2(N) where N is the degree of the irreducible polynomial phi.
+///    The current implementation works always with cc equal to 0b01 and nnnn equal to 0b1001 and thus
+///    the first header by is always equal to 0b00111001.
```

Comment:
We should probably check for this when decoding the signature (in `Polynomial::from_signature()` method) and return an error if the first byte is not what we expect.